addons/udes_stock/tests/test_picking_batch.py

```diff
@@ -1372,6 +1380,32 @@ def test12_computing_partial_assignment(self):
         self.batch01._compute_state()
         self.assertEqual(self.batch01.state, 'done')
 
+    def test13_unreserves_stock_after_cancel_pick(self):
+        """ Cancel pick and check that stock is unreserved"""
+        self.draft_to_ready()
+        self.assign_user()
+        self.assert_number_of_reserved_apples_is(4)
+        self.assertEqual(self.picking01.state, 'assigned', 'Picking 1 should be assigned')
+        self.picking01.do_unreserve()
+        # Cancel the pick and confirm we reach state done
+        self.picking01.action_cancel()
+        self.assert_number_of_reserved_apples_is(0)
+        self.assertEqual(self.batch01.state, 'done', 'Batch 1 should be done')
+        self.assertEqual(self.picking01.state, 'cancel', 'Picking 1 should be cancelled')
+
+    def test14_unreserves_stock_after_unreserve_pick(self):
+        """ Unreserve pick and check that stock is unreserved"""
+        self.draft_to_ready()
+        self.assign_user()
+        self.assert_number_of_reserved_apples_is(4)
+        self.assertEqual(self.picking01.state, 'assigned', 'Picking 1 should be assigned')
+        self.picking01.do_unreserve()
+        # Unreserve the pick and confirm we reach state done
+        self.picking01.do_unreserve()
```

Comment:
do_unreserve() is called twice.

Reply:
yes. I've removed the duplicated call, a similar thing was happening for test13 that I have also fixed